server/commitlog/segment.go

```diff
@@ -383,22 +383,22 @@ func (s *segment) findEntry(offset int64) (e *entry, err error) {
 }
 
 // findEntryByTimestamp returns the first entry whose timestamp is greater than
-// or equal to the given offset.
+// or equal to the given timestamp.
 func (s *segment) findEntryByTimestamp(timestamp int64) (e *entry, err error) {
 	s.RLock()
 	defer s.RUnlock()
 	e = &entry{}
-	n := int(s.Index.Position() / entryWidth)
+	n := int(s.Index.CountEntries())
 	idx := sort.Search(n, func(i int) bool {
-		if err := s.Index.ReadEntryAtFileOffset(e, int64(i*entryWidth)); err != nil {
+		if err := s.Index.ReadEntryAtLogOffset(e, int64(i)); err != nil {
 			panic(err)
 		}
 		return e.Timestamp >= timestamp
 	})
 	if idx == n {
 		return nil, ErrEntryNotFound
 	}
-	err = s.Index.ReadEntryAtFileOffset(e, int64(idx*entryWidth))
+	err = s.Index.ReadEntryAtLogOffset(e, int64(idx))
```

Comment:
Making use of the `ReadEntryAtLogOffset` and `CountEntries` helpers here seemed neater.